aws-api/src/main/java/com/amplifyframework/api/aws/scalar/AWSTemporalTypeAdapter.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.api.aws.scalar;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.TypeAdapter;
+import com.google.gson.TypeAdapterFactory;
+import com.google.gson.reflect.TypeToken;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonToken;
+import com.google.gson.stream.JsonWriter;
+
+import java.io.IOException;
+import java.text.ParseException;
+import java.text.ParsePosition;
+
+/**
+ * Factory to be registered with GSON when dealing with AWS AppSync
+ * scalars. This adapter will correctly de/serialize strings that are
+ * correctly formatted according to the specifications for AWSDate,
+ * AWSTime, and AWSDateTime.
+ *
+ * @see <a href="https://docs.aws.amazon.com/appsync/latest/devguide/scalars.html">AWS AppSync Defined Scalars</a>
+ */
+@SuppressWarnings("ParameterName") // Staying faithful to original method parameter names
+public final class AWSTemporalTypeAdapter implements TypeAdapterFactory {
```

Comment:
If you generic-ize this a bit, you can probably end up with something like:
```
// <T> is AWSDateTime, AWSDate, AWSTime.
final class NullSafeParseAdapter<T> implements TypeAdapterFactory {
    ParseMethod<T> method;
    NullSafeParseAdapter(ParseMethod<T> method) {
        this.method = method;
    }
   @Override
   public <T> TypeAdapter<T> create(Gson gson, TypeToken<T> type) {
       // Same as what you have now, except with type <T> 
       // And in the read() method, call return method.apply(in.nextString())
       // Instead of hard-coding a call to your util
   }
   interface ParseMethod<T> {
       T parse(String string);
   }
}

gson
    .registerTypeAdapterFactory(AWSDateTime.class, new NullSafeParseAdapter(AWSDateTime::fromString))
    .registerTypeAdapterFactory(AWSDate.class, new NullSafeParseAdapter(AWSDate::fromString))
    .registerTypeAdapterFactory(AWSTime.class, new NullSafeParseAdapter(AWSTime::fromString))
```

Reply:
WHOA this is really cool. I had no idea you could do this.